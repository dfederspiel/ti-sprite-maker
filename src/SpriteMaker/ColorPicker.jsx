import React from "react";

const ColorPicker = (props) => {
  const colors = [
    { name: "Black", code: 1, hex: "000000" },
    { name: "Green", code: 2, hex: "21c842" },
    { name: "Light Green", code: 3, hex: "5edc78" },
    { name: "Blue", code: 4, hex: "5455ed" },
    { name: "Light Blue", code: 5, hex: "7d76fc" },
    { name: "Dark Red", code: 6, hex: "d4524d" },
    { name: "Cyan", code: 7, hex: "42ebf5" },
    { name: "Red", code: 8, hex: "fc5554" },
    { name: "Light Red", code: 9, hex: "ff7978" },
    { name: "Dark Yellow", code: 10, hex: "d4c154" },
    { name: "Light Yellow", code: 11, hex: "e6ce80" },
    { name: "Dark Green", code: 12, hex: "21b03b" },
    { name: "Magenta", code: 13, hex: "c95bba" },
    { name: "Gray", code: 14, hex: "cccccc" },
    { name: "White", code: 15, hex: "ffffff" },
  ];

  return (
    <div
      style={{
        display: "flex",
        flexWrap: "wrap",
        justifyContent: "center",
        maxWidth: "400px",
      }}
    >
      {colors.map((color, i) => (
        <div
          key={i}
          style={{
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            fontSize: "16px",
            width: "20px",
            height: "20px",
            padding: "10px",
            margin: "5px",
            backgroundColor: `#${color.hex}`,
          }}
          onClick={() => props.onColorChange(color.hex)}
        >
          {/* {color.name} */}
        </div>
      ))}
    </div>
  );
};

export default ColorPicker;
