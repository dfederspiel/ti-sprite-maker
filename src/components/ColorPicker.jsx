import React from "react";
import styled from "styled-components";

const colors = [
  { name: "Black", code: 1, hex: "000000" },
  { name: "Green", code: 2, hex: "21c842" },
  { name: "Light Green", code: 3, hex: "5edc78" },
  { name: "Blue", code: 4, hex: "5455ed" },
  { name: "Light Blue", code: 5, hex: "7d76fc" },
  { name: "Dark Red", code: 6, hex: "d4524d" },
  { name: "Cyan", code: 7, hex: "42ebf5" },
  { name: "Red", code: 8, hex: "fc5554" },
  { name: "Light Red", code: 9, hex: "ff7978" },
  { name: "Dark Yellow", code: 10, hex: "d4c154" },
  { name: "Light Yellow", code: 11, hex: "e6ce80" },
  { name: "Dark Green", code: 12, hex: "21b03b" },
  { name: "Magenta", code: 13, hex: "c95bba" },
  { name: "Gray", code: 14, hex: "cccccc" },
  { name: "White", code: 15, hex: "ffffff" },
];

const StyledColorTile = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 16px;
  height: 75px;
  background-color: #${(props) => props.hex};
  flex: 1 1 20%;

  @media screen and (min-width: 640px) {
    flex: 0 0 ${100 / colors.length}%;
  }
`;

const ColorPicker = (props) => {
  return (
    <div
      style={{
        display: "flex",
        flexWrap: "wrap",
        justifyContent: "center",
        width: "100%",
      }}
    >
      {colors.map((color, i) => (
        <StyledColorTile
          hex={color.hex}
          key={i}
          onClick={() => props.onColorChange(color.hex)}
        >{color.name}</StyledColorTile>
      ))}
    </div>
  );
};

export default ColorPicker;
