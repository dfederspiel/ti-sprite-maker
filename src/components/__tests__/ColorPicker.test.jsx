import React from 'react';
import { act, fireEvent, render } from '@testing-library/react';
import ColorPicker from '../ColorPicker';

describe('the color picker', () => {
  it('renders as expected', () => {
    const { asFragment } = render(<ColorPicker />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls back with the color changed when a tile is clicked', async () => {
    let counter = 0;
    const { findAllByTestId } = render(<ColorPicker onColorChange={() => {
      // eslint-disable-next-line no-plusplus
      counter++;
    }}
    />);
    const tiles = await findAllByTestId('tile');
    act(() => {
      tiles.forEach((tile) => {
        fireEvent.click(tile);
      });
    });
    expect(counter).toEqual(tiles.length);
  });
});
