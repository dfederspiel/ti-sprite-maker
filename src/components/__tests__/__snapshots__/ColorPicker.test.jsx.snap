// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`the color picker > renders as expected 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-wrap: wrap; justify-content: center; width: 100%;"
  >
    <div
      class="sc-bRKDuR hXlXbx"
      data-testid="tile"
    >
      Black
    </div>
    <div
      class="sc-bRKDuR jWKCBP"
      data-testid="tile"
    >
      Green
    </div>
    <div
      class="sc-bRKDuR colApx"
      data-testid="tile"
    >
      Light Green
    </div>
    <div
      class="sc-bRKDuR cNGPMZ"
      data-testid="tile"
    >
      Blue
    </div>
    <div
      class="sc-bRKDuR iPtQHy"
      data-testid="tile"
    >
      Light Blue
    </div>
    <div
      class="sc-bRKDuR jxorSC"
      data-testid="tile"
    >
      Dark Red
    </div>
    <div
      class="sc-bRKDuR gMNKDf"
      data-testid="tile"
    >
      Cyan
    </div>
    <div
      class="sc-bRKDuR hcAXVd"
      data-testid="tile"
    >
      Red
    </div>
    <div
      class="sc-bRKDuR bTLdGk"
      data-testid="tile"
    >
      Light Red
    </div>
    <div
      class="sc-bRKDuR fkATUq"
      data-testid="tile"
    >
      Dark Yellow
    </div>
    <div
      class="sc-bRKDuR eIIEBk"
      data-testid="tile"
    >
      Light Yellow
    </div>
    <div
      class="sc-bRKDuR jyDSEJ"
      data-testid="tile"
    >
      Dark Green
    </div>
    <div
      class="sc-bRKDuR beUAFP"
      data-testid="tile"
    >
      Magenta
    </div>
    <div
      class="sc-bRKDuR fsjcxV"
      data-testid="tile"
    >
      Gray
    </div>
    <div
      class="sc-bRKDuR gcGxZp"
      data-testid="tile"
    >
      White
    </div>
  </div>
</DocumentFragment>
`;
